Dependencies/Authentication/Sources/Authentication/AuthenticationManager.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  AuthenticationManager.swift
+//
+//
+//  Created by Kamil Wójcicki on 03/10/2023.
+//
+
+import AuthenticationInterface
+import FirebaseAuth
+
+
+final class AuthenticationManager: AuthenticationManagerInterface {
+    
+    init() {
+        userStateListener()
+    }
+    
+    private func userStateListener() {
+        auth.addStateDidChangeListener { [weak self] auth, user in
+            guard let self = self else { return }
+            
+            if let user {
+                self.userID = user.uid
+                self.isAuthenticatedUser = true
+            }
+        }
+    }
+    
+    
+    var userID: String = ""
+    var isAuthenticatedUser: Bool = false
+    private var loginEventHandler: ((Bool) -> Void)?
+    private let auth = Auth.auth()
+    
+    
+    #warning("Not finished yet")
+//    private var userHandler: ((AuthenticationInterface.User?) -> Void)?
+//        lazy var userUpdates: AsyncStream<AuthenticationInterface.User?> = {
+//            AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+//                userHandler = { continuation.yield($0) }
+//            }
+//        }()
+    
+    private var user: AuthenticationInterface.User? {
+        didSet {
+            //userHandler?(user)
+            loginEventHandler?(user != nil)
+        }
+    }
+    
+    lazy var signInResult: AsyncStream<Bool> = {
+        AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+            loginEventHandler = { continuation.yield($0) }
+        }
+    }()
+    
+    func getAuthenticatedUser() throws -> Bool {
+        guard let auth = auth.currentUser else {
+            throw AuthErrorHandler.getAuthenticatedUserError
+        }
+        return true
+    }
+    
+    // MARK: Sign In with email
+    func createUser(email: String, password: String) async throws {
+        do {
+            try await auth.createUser(withEmail: email, password: password)
+        } catch {
+            throw AuthErrorHandler.signUpError
+        }
+    }
+    
+    func signInUser(email: String, password: String) async throws {
+        do {
+            let authDataResult = try await auth.signIn(withEmail: email, password: password).user
+            self.user = User(from: authDataResult)
+        } catch {
+            throw AuthErrorHandler.signInError
+        }
+    }
+    
+    func updatePassword(email: String, password: String, newPassword: String) async throws {
+        guard let user = auth.currentUser else {
+            throw AuthErrorHandler.updatePasswordError
+        }
+        try await user.updatePassword(to: password)
+    }
```

Comment:
```suggestion
    func updatePassword(email: String, password: String, newPassword: String) async throws {
        guard let user = auth.currentUser else {
            throw AuthErrorHandler.getAuthenticatedUserError
        }
        
        do { 
           try await user.updatePassword(to: password)
         } catch {
            throw AuthErrorHandler.updatePasswordError
        }
    }
```

guard let user = auth.currentUser else { throw ... }  sprawdza czy jest user. 

Jeśli go nie ma, to funkcja powinnna wyrzucić błąd który poinformuje użytkownika że jest problem z uwierzytelnieniem.

Czyli w twoim przypadku:
"throw AuthErrorHandler.getAuthenticatedUserError" zamiast 
"throw AuthErrorHandler.updatePasswordError".

Następnie twoja funkcja probuje zaaktualizować hasło i jeśli coś pójdzie nie tak to powinna wyrzucić:
"throw AuthErrorHandler.updatePasswordError"